doc/nrf/protocols/matter/getting_started/custom_clusters.rst

```diff
@@ -0,0 +1,408 @@
+
+   .. code-block::
+
+      west zap-gui -j ./zcl.json ./MyCluster.xml
+
+   This example command copies the original :file:`<default Matter SDK location>/src/app/zap-templates/zcl/zcl.json` file, adds the :file:`MyCluster.xml` cluster, and saves the new :file:`zcl.json` file in the sample directory.
+   The newly generated :file:`zcl.json` file is used as an input to the ZAP tool.
+
+#. Add Endpoint with the new device type in the zap tool.
+
+   .. figure:: images/matter_creating_custom_cluster_new_endpoint.png
+      :alt: Endpoint with My new device in ZAP tool
+
+      Endpoint with My new device in ZAP tool
+
+#. Locate the new cluster in the ZAP tool.
+
+   .. figure:: images/matter_creating_custom_cluster_new_cluster.png
+      :alt: New custom cluster in ZAP tool
+
+      New custom cluster in ZAP tool
+
+#. Choose whether the cluster should be enabled for the Client and Server sides.
+
+#. Click the gear icon to open the cluster configuration and enable the attributes, commands, and events.
+
+   a. In the :guilabel:`ATTRIBUTES` tab, ensure that you have the required attributes enabled.
+
+      .. figure:: images/matter_creating_custom_cluster_attributes.png
+         :alt: Attributes of the new custom cluster in ZAP tool
+
+         Attributes of the new custom cluster in ZAP tool
+
+   #. In the :guilabel:`COMMANDS` tab, ensure that you have the required commands enabled.
+
+      .. figure:: images/matter_creating_custom_cluster_commands.png
+         :alt: Commands of the new custom cluster in ZAP tool
+
+         Commands of the new custom cluster in ZAP tool
+
+   #. In the :guilabel:`EVENTS` tab, ensure that you have the required events enabled.
+
+      .. figure:: images/matter_creating_custom_cluster_events.png
+         :alt: Events of the new custom cluster in ZAP tool
+
+         Events of the new custom cluster in ZAP tool
+
+#. Save the file and exit.
+
+.. rst-class:: numbered-step
+
+Generate the C++ code that contains the selected clusters
+*********************************************************
+
+Run the following command to use the modified ZAP file to generate the C++ code that contains the selected clusters:
+
+   .. code-block::
+
+      west zap-generate --full
+
+After completing these steps, the following changes will be visible within your sample directory:
+
+* The new cluster description file :file:`MyCluster.xml`.
+* The updated data model definition file :file:`zcl.json` with the new cluster and relative paths to the Matter data model directory.
+* The generated C++ source files for the new cluster.
+* The updated :file:`.zap` file with the new cluster configuration and relative path to the :file:`zcl.json` file.
+
+Once the new cluster is added to the Matter application, you can call the ``zap-gui`` command without the additional ``--clusters`` argument.
+However, you still need to provide the path to the :file:`zcl.json` file if you created a new one in a different location than the default one.
```

Comment:
Let's inform here that the .zap gui will contain a path to the new zcl.json file, hence there is no need to provide it once again by an argument to the zap-gui command.

Reply:
will ad a note

Reply:
as discussed offline: current implementation requires providing `-j` flag always when using custom zcl.json . Added comment to improve that in `KRKNWK-20040`